ldp/nn/handlers/transformer_handler.py

```diff
@@ -0,0 +1,1017 @@
+        os.environ["CUDA_VISIBLE_DEVICES"] = str(self.local_rank)
+        os.environ["RANK"] = str(self.rank)
+        os.environ["WORLD_SIZE"] = str(self.world_size)
+        os.environ["LOCAL_RANK"] = str(self.local_rank)
+        os.environ["MASTER_ADDR"] = self.master_addr
+        os.environ["MASTER_PORT"] = str(self.master_port)
+
+        # These tell PretrainedModel.from_pretrained() that we're using FSDP
+        os.environ["ACCELERATE_USE_FSDP"] = "1"
+        os.environ["FSDP_CPU_RAM_EFFICIENT_LOADING"] = str(
+            int(self.cpu_ram_efficient_loading)
+        )
+
+
+class ParallelTransformerHandler(TransformerHandler):
+    def __init__(
+        self,
+        config: TransformerHandlerConfig,
+        parallel_worker_config: ParallelWorkerConfig,
+    ):
+        parallel_worker_config.set_env_vars()
+        dist.init_process_group(backend="nccl")
+        self.worker_config = parallel_worker_config
+        super().__init__(config)
+
+    def _setup_accelerator(self):
+        bf16 = self.config.lm_config.dtype == TorchDType.bf16
+
+        mixed_precision = None
+        if bf16:
+            bf16_ready = (
+                torch.version.cuda
+                and torch.cuda.is_bf16_supported()
+                and torch.version.cuda >= "11.0"
+                and dist.is_nccl_available()
+                and nccl.version() >= (2, 10)
+            )
+            assert bf16_ready, (
+                "Mixed precision training requires CUDA 11.0 and NCCL 2.10"
+            )
+            mixed_precision = MixedPrecision(
+                param_dtype=torch.bfloat16,
+                reduce_dtype=torch.bfloat16,
+                buffer_dtype=torch.bfloat16,
+            )
+
+        self.accelerator = accelerate.Accelerator(
+            # See note in TransformerHandler._setup_accelerator() about this
+            # mixed_precision=("bf16" if bf16 else "no"),
+            fsdp_plugin=accelerate.FullyShardedDataParallelPlugin(
+                sharding_strategy=ShardingStrategy.FULL_SHARD,
+                mixed_precision_policy=mixed_precision,
+                auto_wrap_policy="transformer_based_wrap",
+                cpu_offload=self.worker_config.offload_cpu,
+                activation_checkpointing=self.worker_config.activation_checkpointing,
+                cpu_ram_efficient_loading=self.worker_config.cpu_ram_efficient_loading,
+                sync_module_states=self.worker_config.cpu_ram_efficient_loading,
+                state_dict_type=self.worker_config.state_dict_type,
+                backward_prefetch=self.worker_config.backward_prefetch,
+            ),
+        )
+
+        if self.config.lm_config.device == "meta":
+            self.module = prepare_model_for_fsdp_with_meta_device(self.module)
+
+        self.module = self.accelerator.prepare(self.module)
+
+    def set_seed(self, seed: int) -> None:
+        """Set the seed for the current worker."""
+        set_seed(seed)
```

Comment:
Do we need this method? Can we just have the caller import/use `set_seed` themselves?